lib/pull-request/merge-back.ts

```diff
@@ -37,7 +37,12 @@ export interface MergeBackStage {
   readonly after: string;
 }
 
-export interface AutoMergeBackOptions extends pr.AutoPullRequestOptions {
+/**
+ * AutoMergeBackOptions and AutoMergeBackProps both share all of these options,
+ * but AutoMergeBackProps can't extend AutoMergeBackOptions; the latter has a 'stage' prop the former doesn't have.
+ * This interface instead captures all of the common props between the two.
+ */
+interface CommonMergeOptions extends pr.AutoPullRequestOptions {
```

Comment:
How about something like this?

AutoMergeBackProps will extend AutoMergeBackOptions, neither of which will have stage.
Another struct, say StageMergeBackOption extends AutoMergeBackOptions and includes stage.

A few things become cleaner this way - this CommomMergeBackOptions goes away which sounds awkward. And AutoMergeBackProps extends AutoMergeBackOptions which is more natural in how we model construct props.

Will that work?

Reply:
I like this -- and considered something similar -- except it removes `stage` from the `AutoMergeBackOptions` interface, which would be backwards-incompatible. Now practically, as a JS/TS user of this library, it would largely be invisible (as long as the pipelines interface is changed to a `StageMergeBackOptions`), but it does seem like it would require a major version bump to be semver-compliant. Meanwhile, the point of this change was to fix a backwards-incompatible change which broke semver for our consumers, so doing another backwards-incompatible change here feels wrong.

(After discussing offline) - Let's ship this as-is, then do your suggestion as part of a major version bump.